pkg/homedir/homedir.go

```diff
@@ -21,11 +25,33 @@ func GetConfigHome() (string, error) {
 	return filepath.Join(home, ".config"), nil
 }
 
-// GetDataHome returns XDG_DATA_HOME.
+// GetDataHome returns XDG_DATA_HOME for the given UID
 // GetDataHome returns $HOME/.local/share and nil error if XDG_DATA_HOME is not set.
 //
 // See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
-func GetDataHome() (string, error) {
+func GetDataHome(uid int) (string, error) {
```

Comment:
Add a new function
GetDataHomeByUID(uid int) ...

And have GetDataHome call it.